serialize.go

```diff
@@ -32,6 +32,39 @@ func InOrderSerialize(tree *Tree) []NodeData {
 	return res
 }
 
+func StableSerializeBFS(tree *Tree) []NodeData {
+	if tree.root == nil {
+		return nil
+	}
+
+	size := tree.Size()
+	visited := map[string][]byte{}
+	keys := make([][]byte, 0, size)
+	numKeys := -1
+
+	// Breadth-first search. At every depth, add keys in search order. Keep
```

Comment:
This seems fine, but depends on the meaning of the key stored in an inner node.
Please define what that is here to make the algorithm understandable.

(aka key of an inner node, is the key of the leftmost leaf of the right child)

Reply:
True, is this better: https://medium.com/speaking-human/welcome-to-the-dream-society-5ac1768e3cda ? Since AVL trees by design have this property where inner nodes are ordered.